load-testing/tests/relays_stress_helpers_test.go

```diff
@@ -115,22 +116,33 @@ func (s *relaysSuite) initFundingAccount(fundingAccountKeyName string) {
 	}
 }
 
-// initializeProvisionedActors parses the load test manifest and initializes the
+// initializeLoadTestParams parses the load test manifest and initializes the
 // gateway and supplier keyNames and the URLs used to send requests to.
-func (s *relaysSuite) initializeProvisionedActors() {
-	loadTestManifestContent, err := os.ReadFile(loadTestManifestPath)
+func (s *relaysSuite) initializeLoadTestParams() *config.LoadTestManifestYAML {
+	loadTestManifestContent, err := os.ReadFile(flagManifestFilePath)
 	require.NoError(s, err)
 
-	provisionedActors, err := config.ParseLoadTestManifest(loadTestManifestContent)
+	loadTestManifest, err := config.ParseLoadTestManifest(loadTestManifestContent)
 	require.NoError(s, err)
 
-	for _, gateway := range provisionedActors.Gateways {
-		s.gatewayUrls[gateway.KeyName] = gateway.ExposedUrl
+	s.persistentChain = loadTestManifest.PersistentChain
+
+	for _, gateway := range loadTestManifest.Gateways {
+		// In the case of persistent chain, the test has no entry for the gateway
+		// and no keyName is provided. The gateway address is used to identify the
+		// gateway in the network.
+		if s.persistentChain {
```

Comment:
1. The code is going to get really messy with all of these ifs all over the place
2. There is already an implicit assumption that we should only define `Address` or `KeyName` but not both
3. I think this means we may need:
- Mutiple structs
- Multiple functions

It gets messy when you star doing this (and then adding another bool, and another)

Reply:
We considered having multiple functions but they will be almost duplicated (except for what's in these conditionals).

I propose to keep but document them. And refactor if/when we introduce more functionality in the future.